src/Models/BaseElement.php

```diff
@@ -679,6 +679,38 @@ public function getAnchor()
         return $this->anchor = $result;
     }
 
+    /**
+     * Get anchors in this block's content.
+     * Used to populate the "anchor on a page" link in the WYSIWYG
+     *
+     * By default, this finds anchors in any HTMLText field on the block, but
+     * this method should be overridden if anchors are provided in other ways
+     * for this block or if not all HTMLText fields for this block are
+     * displayed on the front-end.
+     */
+    public function getAnchorsInContent(): array
+    {
+        $anchors = [$this->getAnchor()];
+        $anchorRegex = "/\\s+(name|id)\\s*=\\s*([\"'])([^\\2\\s>]*?)\\2|\\s+(name|id)\\s*=\\s*([^\"']+)[\\s +>]/im";
```

Comment:
I used https://regex101.com/ for testing regular expression and I found that second part of this condition doesn't make sence. It looks like it searches only for quotes. I would suggest to remove this part or modify it for some special purpose.

Reply:
Can you please specify which part you mean by "the second part"?
I took this regex straight from [`SiteTree::getAnchorsOnPage()`](https://github.com/silverstripe/silverstripe-cms/blob/77b5bfa497d5026ffc806f1071a6c5d04d65bfb4/code/Model/SiteTree.php#L3337) so if it's flawed we'll need to update it there too.

Reply:
How I can see we have two conditions:
It is looking for either `\s+(name|id)\s*=\s*([\"'])([^\2\s>]*?)\2`(first part)  or  `\s+(name|id)\s*=\s*([^\"']+)[\s +>]`(second part)
If you copy second part and test it in https://regex101.com/, use this string for test `<div><p id="my_p_link">My paragraph</p></div>`, it cannot find any matches. Probably, this regexpression should test some other sort of string.

Reply:
Right. So that second section is looking at markup where the quotes have been missed out, e.g. `<a name=anchor-name>`. this was taken from the existing `SiteTree::getAnchorsOnPage()` so I feel like omitting or changing it is likely to cause regressions.

Reply:
Don't worry. It was just a suggestion. Probably, we should just do more tests and enchance it.